test/unit/org/apache/cassandra/tools/StandaloneSplitterWithCQLTesterTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.tools;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.io.Files;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.apache.cassandra.OrderedJUnit4ClassRunner;
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.io.sstable.format.SSTableReader;
+import org.apache.cassandra.tools.ToolRunner.ToolResult;
+import org.assertj.core.api.Assertions;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(OrderedJUnit4ClassRunner.class)
+public class StandaloneSplitterWithCQLTesterTest extends CQLTester
+{
+    private String sstableFileName;
+    private File sstablesDir;
+    private File sstablesBackupDir;
+    private List<File> origSstables;
+
+    // CQLTester post test method cleanup needs to be avoided by putting all tests in single junit test method
+
+    @Test
+    public void test() throws Throwable
```

Comment:
Can you break down the test cases into individual tests? Therefore, CI continues testing the other case on failure. It may reveal a more accurate result. 

You can tell the test runner to ignore the annotations from `@Before`, `@After`, `@BeforeClass`, etc. by extending the `OrderedJUnit4ClassRunner` and overriding `withBefore()`... methods.  
So it also solves the problem that you immediately `tearDownClass` in the test.

Reply:
Good thinking. Please check the latest commit :-)